src/server/game/Movement/PathGenerator.cpp

```diff
@@ -904,7 +904,7 @@ float PathGenerator::Dist3DSqr(G3D::Vector3 const& p1, G3D::Vector3 const& p2) c
     return (p1 - p2).squaredLength();
 }
 
-void PathGenerator::ReducePathLenghtByDist(float dist)
+void PathGenerator::ReducePathLenghtByDist(const float dist, Unit* target)
```

Comment:
It would be nice if target wouldn't be mandatory (right now the code crashes with NULL dereference exception of the caller passes NULL since there's no validation at all), PathGenerator is not supposed to always have a Unit as target.

Reply:
please add an ASSERT(target); here